sdk/ml/azure-ai-ml/azure/ai/ml/operations/_featureset_operations.py

```diff
@@ -0,0 +1,227 @@
+# ---------------------------------------------------------
+
+# pylint: disable=protected-access
+
+from typing import Dict, Optional
+
+from marshmallow.exceptions import ValidationError as SchemaValidationError
+
+from azure.ai.ml._exception_helper import log_and_raise_error
+from azure.ai.ml._restclient.v2023_02_01_preview.models import ListViewType, FeaturesetVersion
+from azure.ai.ml._restclient.v2023_02_01_preview import AzureMachineLearningWorkspaces as ServiceClient022023Preview
+from azure.ai.ml._scope_dependent_operations import OperationConfig, OperationScope, _ScopeDependentOperations
+from azure.ai.ml.exceptions import ErrorCategory, ErrorTarget, ValidationErrorType, ValidationException
+
+
+# from azure.ai.ml._telemetry import ActivityType, monitor_with_activity
+from azure.ai.ml._utils._asset_utils import (
+    _archive_or_restore,
+    _get_latest_version_from_container,
+    _resolve_label_to_asset,
+)
+from azure.ai.ml._utils._logger_utils import OpsLogger
+from azure.ai.ml.entities._assets import Featureset
+from azure.ai.ml._utils._experimental import experimental
+from azure.core.polling import LROPoller
+from azure.core.paging import ItemPaged
+
+ops_logger = OpsLogger(__name__)
+module_logger = ops_logger.module_logger
+
+
+@experimental
+class FeaturesetOperations(_ScopeDependentOperations):
+    def __init__(
+        self,
+        operation_scope: OperationScope,
+        operation_config: OperationConfig,
+        service_client: ServiceClient022023Preview,
+        **kwargs: Dict,
+    ):
+
+        super(FeaturesetOperations, self).__init__(operation_scope, operation_config)
+        # ops_logger.update_info(kwargs)
+        self._operation = service_client.featureset_versions
+        self._container_operation = service_client.featureset_containers
+        self._service_client = service_client
+        self._init_kwargs = kwargs
+
+        # Maps a label to a function which given an asset name,
+        # returns the asset associated with the label
+        self._managed_label_resolver = {"latest": self._get_latest_version}
+
+    # @monitor_with_activity(logger, "Featureset.List", ActivityType.PUBLICAPI)
+    def list(
+        self,
+        *,
+        name: Optional[str] = None,
+        list_view_type: ListViewType = ListViewType.ACTIVE_ONLY,
+    ) -> ItemPaged[Featureset]:
+        """List the featureset assets of the workspace.
+
+        :param name: Name of a specific featureset asset, optional.
+        :type name: Optional[str]
+        :param list_view_type: View type for including/excluding (for example) archived featureset assets.
+        Default: ACTIVE_ONLY.
+        :type list_view_type: Optional[ListViewType]
+        :return: An iterator like instance of Featureset objects
+        :rtype: ~azure.core.paging.ItemPaged[Featureset]
+        """
+        if name:
+            return self._operation.list(
+                workspace_name=self._workspace_name,
```

Comment:
Do featureset operations only make sense in the scope of a feature store workspace? If so, should this be feature_store_name?

Reply:
the ml client don't have property for feature_store_name. 
The ml client constructor takes the feature store as worspace name, since feature store is also a kind of workspace.